src/dev/run_extract_default_translations.js

```diff
@@ -0,0 +1,343 @@
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+const makeDir = promisify(fs.mkdir);
+const globProm = promisify(glob);
+
+run(async () => {
+  for (const inputPath of process.argv.slice(2)) {
+    await extractDefaultTranslations(inputPath);
+  }
+});
+
+async function isPathExists(path) {
+  await promisify(fs.access)(path, fs.constants.F_OK);
+}
+
+function addMessageToMap(targetMap, key, value) {
+  if (targetMap.has(key) && targetMap.get(key) !== value) {
+    throw new Error(`Default messages are different for id "${key}"`);
+  }
+  targetMap.set(key, value);
+}
+
+function parseConditionalOperatorAST(subTree) {
+  const ids = [];
+
+  if (babelTypes.isStringLiteral(subTree)) {
+    ids.push(subTree.value);
+  } else if (babelTypes.isConditionalExpression(subTree)) {
+    ids.push(...parseConditionalOperatorAST(subTree.consequent));
+    ids.push(...parseConditionalOperatorAST(subTree.alternate));
+  }
+
+  return ids;
+}
+
+function parseConditionalOperatorExpression(expression) {
+  let ids = [];
+
+  traverse(parse(expression), {
+    enter(path) {
+      if (babelTypes.isDirectiveLiteral(path.node)) {
+        ids = [path.node.value];
+        path.stop();
+      } else if (babelTypes.isConditionalExpression(path.node)) {
+        ids = parseConditionalOperatorAST(path.node);
+        path.stop();
+      }
+    },
+  });
+
+  return ids;
+}
+
+function parseFilterObjectExpression(expression) {
+  let defaultMessage = '';
+
+  const filterObjectAST = parse(`+${expression}`);
+  traverse(filterObjectAST, {
+    enter(path) {
+      if (babelTypes.isObjectExpression(path.node)) {
+        path.node.properties.forEach(property => {
+          if (isPropertyWithKey(property, defaultMessageKey) && babelTypes.isStringLiteral(property.value)) {
+            defaultMessage = property.value.value;
+          }
+        });
+      }
+    }
+  });
+
+  return defaultMessage;
+}
+
+function getFilterMessages(htmlContent) {
+  const expressions = (htmlContent.match(angularExpressionRegEx) || [])
+    .filter(expression => expression.indexOf('|') !== -1)
+    .filter(expression => expression.indexOf('i18n') !== -1)
+    .map(expression => expression.slice(2, -2).trim());
+
+  const defaultMessagesMap = new Map();
+
+  expressions.forEach(expression => {
+    const [idsExpression] = expression.match(/[\s\S]+(?=\|[\s]*i18n)/g) || [];
+    const [filterObjectExpression] = expression.match(/(?<=\|[\s]*i18n[\s]*:)[\s\S]+/g) || [];
+
+    if (!filterObjectExpression || !idsExpression) {
+      throw new Error('Angular.js expression syntax error');
+    }
+
+    const messagesIds = parseConditionalOperatorExpression(idsExpression);
+    const messageValue = parseFilterObjectExpression(filterObjectExpression);
+
+    messagesIds.forEach(id => {
+      addMessageToMap(defaultMessagesMap, id, messageValue);
+    });
+  });
+
+  return defaultMessagesMap;
+}
+
+function getMessagesIdsFromHTMLElement(attributeValue) {
```

Comment:
__question:__ Could you please clarify what you're doing here exactly, in what case we can have more than one id (and would be great to have this explanation in JSDoc for this function)?

Reply:
In case of conditional expression:
`i18n-id="{{ condition ? 'ID_1' : 'ID_2' }}"`

Reply:
Got it, hopefully we won't do this (and this `i18n-id="{{ some_generated_id }}"`) in real life (I'd prefer to disable this with custom ESLint/TSLint check ideally, we can get back to this later if it becomes a problem). Please mention this in JSDoc for this function as well.